frontend/src/components/auth/Password/PasswordForm.tsx

```diff
@@ -0,0 +1,197 @@
+        <FormError message="Please ensure passwords match" />
+      )}
+      {displayRequirementError && (
+        <FormError message="Password does not meet all of the requirements" />
+      )}
+      {version === "AdminSignup" && (
+        <FormControl isRequired pb={6}>
+          <FormLabel color="grey.400">Email Address</FormLabel>
+          <Input type="email" value={email} disabled />
+        </FormControl>
+      )}
+      <FormControl isRequired pb={6}>
+        <FormLabel color="grey.400">Password</FormLabel>
+        <Input
+          type="password"
+          placeholder="Enter Password"
+          onChange={handlePasswordChange}
+          isInvalid={hasError}
+        />
+      </FormControl>
+      <FormControl isRequired pb={6}>
+        <FormLabel color="grey.400">Confirm Password</FormLabel>
+        <Input
+          type="password"
+          placeholder="Enter Password"
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            setConfirmPassword(e.target.value)
+          }
+          isInvalid={hasError}
+        />
+      </FormControl>
+      <HStack alignItems="top" pb={6}>
+        <VStack alignItems="left" mr={4}>
+          <PasswordRequirement
+            isFulfilled={minNineChars}
+            requirement="minimum of 9 characters"
+          />
+          <PasswordRequirement
+            isFulfilled={hasUppercase}
+            requirement="one uppercase letter"
+          />
+          <PasswordRequirement
+            isFulfilled={hasLowercase}
+            requirement="one lowercase letter"
+          />
+        </VStack>
+        <VStack alignItems="left">
+          <PasswordRequirement
+            isFulfilled={hasSpecial}
+            requirement="one special character"
+          />
+          <PasswordRequirement isFulfilled={hasNum} requirement="one number" />
+        </VStack>
+      </HStack>
+      {version === "AdminSignup" && (
+        <Button
+          variant="primary"
+          width="100%"
+          onClick={(e: React.MouseEvent<HTMLButtonElement>) => onClick(e)}
+        >
+          Create Account
+        </Button>
+      )}
+      {version === "ResetPassword" && (
+        <NavigationButtons
+          onContinueClick={(e: React.MouseEvent<HTMLButtonElement>) =>
+            onClick(e)
+          }
+          onBackClick={() =>
+            history.push(userRole === "Admin" ? ADMIN_LOGIN : TEACHER_LOGIN)
+          }
+          continueButtonText="Reset Password"
+          backButtonText="Back to login page"
+        />
+      )}
+      {version === "TeacherSignup" && (
+        <NavigationButtons
+          onContinueClick={(e: React.MouseEvent<HTMLButtonElement>) =>
+            onClick(e)
+          }
+          onBackClick={() => setStep && setStep(3)} // TODO: Back button can either go to page 2 or 3 depending on whether school exists or not
+          continueButtonText="Continue"
+          backButtonText="Back"
+        />
```

Comment:
hmm i think the reason i did this was to have one `onClick()` function so i could define all the useState variables within `PasswordForm` instead of the individual components. if we were to break it up, there would be essentially no point of having this shared `PasswordForm` component since we would need to pass like 18 variables as props if we were to just render the actual form part (which is identical btwn versions)

so, other option is to just define everything in their separate components and get rid of this file - i guess this is tradeoff btwn having some repetition vs the "version" stuff

Reply:
ohhh okay! sounds good